### PR TITLE
Revamped Upgrade Agent for Litmus 3.x.x 

### PR DESCRIPTION
**Pull request by kartikaysaxena**
## Proposed changes

Related #4632 

Summarize your changes here to communicate with the maintainers and make sure to put the link of that issue

## Types of changes

What types of changes does your code introduce to Litmus? Put an `x` in the boxes that apply
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices applies)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
- [ ] I have read the [CONTRIBUTING](https://github.com/litmuschaos/litmus/blob/master/CONTRIBUTING.md) doc
- [ ] I have signed the commit for DCO to be passed.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have added necessary documentation (if appropriate)

## Dependency
- Please add the links to the dependent PR need to be merged before this (if any).

## Special notes for your reviewer:
Revamped Litmus Upgrade Agent for Litmus 3.x.x

### CODE REVIEW

**Review by SarthakJain26 — COMMENTED** [review]
Please remove the commented code

File: chaoscenter/upgrade-agents/control-plane/versions/upgrade-manager.go [hunk 1]
```diff
@@ -158,55 +214,133 @@ func (m *UpgradeManager) getUpgradePath() map[string]UpgradeExecutor {
...
+		},
+		"3.7.7-beta8": {
+			NextVersion:    "",
+			VersionManager: nil,
+		},
+		"3.9.0": {
+			NextVersion:    "",
+			VersionManager: v3_9_0.NewVersionManger(m.Logger, m.DBClient),
+		},
+		"4.0.0-beta8": {
```
Comment (Saranya-jena) [inline chaoscenter/upgrade-agents/control-plane/versions/upgrade-manager.go:230]: We can remove this for now

**Review by Saranya-jena — APPROVED** [review]

File: chaoscenter/upgrade-agents/control-plane/versions/upgrade-manager.go [hunk 2]
```diff
@@ -158,55 +214,129 @@ func (m *UpgradeManager) getUpgradePath() map[string]UpgradeExecutor {
 			NextVersion:    "",
 			VersionManager: nil,
 		},
+
+		"3.6.0": {
+			NextVersion:    "3.7.0",
+			VersionManager: nil,
+		},
+		"3.7.7-beta8": {
```
Comment (SarthakJain26) [inline chaoscenter/upgrade-agents/control-plane/versions/upgrade-manager.go:222]: All these old versions where no upgrades are available can be removed from this map

File: chaoscenter/upgrade-agents/control-plane/versions/upgrade-manager.go [hunk 3]
```diff
@@ -158,55 +214,129 @@ func (m *UpgradeManager) getUpgradePath() map[string]UpgradeExecutor {
...
+			VersionManager: nil,
+		},
+		"3.9.0": {
+			NextVersion:    "",
+			VersionManager: v3_9_0.NewVersionManger(m.Logger, m.DBClient),
+		},
+	}
+}
+
+func arrayToVersion(versionArray []int) string {
```
Comment (SarthakJain26) [inline chaoscenter/upgrade-agents/control-plane/versions/upgrade-manager.go:233]: Please add a comment to describe what this function does

File: chaoscenter/upgrade-agents/control-plane/versions/v0.0.0/upgrade.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+	"github.com/litmuschaos/litmus/chaoscenter/upgrader-agents/control-plane/pkg/database"
+	log "github.com/sirupsen/logrus"
+)
+
+/* This example upgrade renames auth collection users to client*/
+
+func upgradeUsersCollection(logger *log.Logger, dbClient *mongo.Client, ctx context.Context) error {
+
+	logVersion := log.Fields{
```
Comment (SarthakJain26) [inline chaoscenter/upgrade-agents/control-plane/versions/v0.0.0/upgrade.go:19]: Can you please follow this kind of logging here: https://github.com/litmuschaos/litmus/pull/4740/files#diff-c3dcc200129e5965fb7dc2655155eb285cfea424297940f911ffe1b6624b8316R62

File: chaoscenter/upgrade-agents/control-plane/versions/v0.0.0/upgrade.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+	cursor, err := indexView.List(ctx)
+	if err != nil {
+		log.Error("error listing indexes: %w", err)
+		return err
+	}
+	defer cursor.Close(ctx)
+
+	environmentIDIndexExists := false
+
+	logVersion := log.Fields{
```
Comment (SarthakJain26) [inline chaoscenter/upgrade-agents/control-plane/versions/v0.0.0/upgrade.go:54]: Can you please follow this kind of logging here: https://github.com/litmuschaos/litmus/pull/4740/files#diff-c3dcc200129e5965fb7dc2655155eb285cfea424297940f911ffe1b6624b8316R62

Please update in other places as well

File: chaoscenter/upgrade-agents/control-plane/versions/v3.9.0/manager.go [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+// Run executes all the steps required for the Version Manger
+// to upgrade from the previous version to `this` version
+func (vm VersionManager) Run() error {
+	ctx := context.Background()
+	session, err := vm.DBClient.StartSession()
+
+	defer session.EndSession(ctx)
+
+	logVersion := log.Fields{
```
Comment (SarthakJain26) [inline chaoscenter/upgrade-agents/control-plane/versions/v3.9.0/manager.go:29]: Can you please follow this kind of logging here: https://github.com/litmuschaos/litmus/pull/4740/files#diff-c3dcc200129e5965fb7dc2655155eb285cfea424297940f911ffe1b6624b8316R62

**Review by SarthakJain26 — APPROVED** [review]
Great work @kartikaysaxena 🚀 , This is a great feature addition and will help the users in seamless upgrade of chaos center.

**Comment by Saranya-jena** [comment]
Awesome work @kartikaysaxena 🚀 ❤️

**Pull request merged**